cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/GcpMetadataHttpHandlerInternal.java

```diff
@@ -89,6 +95,18 @@ public GcpMetadataHttpHandlerInternal(CConfiguration cConf,
         new GcpWorkloadIdentityInternalAuthenticator(gcpMetadataTaskContext);
     this.credentialProvider = new RemoteNamespaceCredentialProvider(remoteClientFactory,
         this.gcpWorkloadIdentityInternalAuthenticator);
+    this.credentialLoadingCache = CacheBuilder.newBuilder()
```

Comment:
If I understand correctly, this loading cache will only ever have one element in it?

Reply:
It might not be the case always, during my testing in case of multiple GCP plugins, for eg, `BQ` & `GCS` will generate credentials with different scopes.

Reply:
SG, thank you!